lib/src/sdk_api/view/video_view.dart

```diff
@@ -10,36 +9,49 @@ import 'package:flutter/widgets.dart';
 
 import 'package:dolbyio_comms_sdk_flutter/dolbyio_comms_sdk_flutter.dart';
 
-/// A controller for the [VideoView] that is responsible for attaching a [Participant] and a [MediaStream] to the [VideoView], detaching them, and getting information about the [VideoView] state.
+/// A controller for the [VideoView] that is responsible for attaching a [Participant] and a
+/// [MediaStream] to the [VideoView], detaching them, and getting information about the
+/// [VideoView] state.
 /// 
 /// An instance of this class can be provided during the instantiation of the [VideoView] widget, 
 /// if this widget is constructed using [VideoView.forList].
 class VideoViewController {
 
   _VideoViewState?  _state;
+
+  Participant? _participant;
+  MediaStream? _mediaStream;
   
   /// Attaches a [Participant] and a [MediaStream] to the [VideoView]. This allows the
   /// [VideoView] to display the provided [MediaStream] if the media stream object belongs
   /// to the provided [Participant]. 
-  Future<void> attach(Participant participant, MediaStream? mediaStream) {
+  void attach(Participant participant, MediaStream? mediaStream) {
+
+    _participant = participant;
+    _mediaStream = mediaStream;
+
     final state = _state;
-    if (state != null) {
-      return state._attach(participant, mediaStream);
+    if (state != null && state._methodChannel != null) {
```

Comment:
bad design, view controller cannot be responsible to check if some field inside other object equals null. (single responsibility principle)